### PR TITLE
Feature/mendixwrapper

### PR DESCRIPTION
**Pull request by djkwagala**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+Try our the demo project https://rating100.mxapps.io with username: x/y/z  and password: 1
+Editing of a rate is possible if the logged in user owns that vote
+
+## Usage
+This widget requires a context.
+
+### Security Configuration
+
+Security is a mandatory feature for rating and is considered seriously for proper use of this widget in a mendix project.
+ - Model Configuration ![1](./assets/domain_model.jpg)
```
Comment (Andries-Smit) [inline README.md]: Only use full url's https://github.com/FlockOfBirds/star-rating/blob/feature/mendixwrapper/assets/domain_model.jpg

Else the App store documentation URL's will break.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ - Campaign entity security: a User should only read the average attribute ![1](./assets/security_campaign.jpg)
+ - Rating or voting entity security: The user can only write their own rating ![1](./assets/security_rate.jpg)
+ - When creating a new rating, its important to use the current user's previous rating as the initial rate value. use this microflow [RateMe microflow](https://modelshare.mendix.com/models/d7ece331-49d4-4464-a2e2-ea75528a0367/rate-me) for the rate me custom button
+ - Calculation of the average rate is done by adding a [Calculate average microflow](https://modelshare.mendix.com/models/d27114b6-e2fb-4d79-aa39-8c60a6477ca8/calculate-average-rate) to the after commit and and after delete event handlers in the domain modeler
+
+## Issues
```
Comment (Andries-Smit) [inline README.md]: Redundant, please remove

File: src/StarRating.xml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            <caption>Value attribute</caption>
+            <category>General</category>
+            <description>The attribute containing a rating</description>
+            <attributeTypes>
+                <attributeType name="Decimal"/>
+                <attributeType name="Integer"/>
+                <attributeType name="Long"/>
+            </attributeTypes>
+        </property>
+
```
Comment (Andries-Smit) [inline src/StarRating.xml]: Remove newline

File: src/components/StarRatingContainer.ts [hunk 4]
```diff
@@ -0,0 +1,127 @@
+import { Component, createElement } from "react";
+import { StarRating } from "./StarRating";
+
+interface WrapperProps {
+    class?: string;
+    mxObject?: mendix.lib.MxObject;
+    style?: string;
+}
```
Comment (Andries-Smit) [inline src/components/StarRatingContainer.ts]: Please check if `className `and `style` will always be there, I think they well be an empty string so in interface does not need be optional. Remove `?`

File: src/components/StarRatingContainer.ts [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+export interface ContainerProps extends WrapperProps {
+    editable: "default" | "never";
+    mxObject: mendix.lib.MxObject;
+    viewAverage: boolean;
+    readOnly: boolean;
```
Comment (Andries-Smit) [inline src/components/StarRatingContainer.ts]: `readOnly `and `mxObject` are redundant, as they are already extended from `WrapperProps`, please remove

File: src/components/__tests__/StarRating.spec.ts [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+        );
+    });
+
+    it("should render with whole numbers for the rating rate type", () => {
+        const starRating = renderStarRating(starProps).find(Rating);
+
+        expect(starRating.props().fractions).toBe(1);
+    });
+
+    it("should render with fractions for the average rate type", () => {
```
Comment (Andries-Smit) [inline src/components/__tests__/StarRating.spec.ts]: `average` is not any longer part of the widget. its when readOnly.

File: src/components/StarRatingContainer.ts [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+            return style.split(";").reduce<{ [key: string]: string }>((styleObject, line) => {
+                const pair = line.split(":");
+                if (pair.length === 2) {
+                    const name = pair[0].trim().replace(/(-.)/g, match => match[1].toUpperCase());
+                    styleObject[name] = pair[1].trim();
+                }
+                return styleObject;
+            }, {});
+        } catch (error) {
+            console.log("Failed to parse style", style, error);
```
Comment (Andries-Smit) [inline src/components/StarRatingContainer.ts]: better not use console use `window.logger.error`

File: src/components/StarRatingContainer.ts [hunk 8]
```diff
@@ -0,0 +1,127 @@
...
+    }
+
+    render() {
+        const { mxObject } = this.props;
+        const readOnly = this.props.editable === "never"
+            || (mxObject && mxObject.isReadonlyAttr(this.props.rateAttribute)) || this.props.readOnly || !mxObject;
+
+        return createElement(StarRating, {
+            className: this.props.class,
+            fractions: readOnly ? 2 : 1,
```
Comment (Andries-Smit) [inline src/components/StarRatingContainer.ts]: Fractions should not be part of the container logic; it is display logic, only mx interactions should be in the container. Move this settings to the _StarRating.ts_

File: README.md (hunk 1 above)
Comment (djkwagala) [inline README.md]: Fixed

File: README.md (hunk 2 above)
Comment (djkwagala) [inline README.md]: removed

File: src/StarRating.xml (hunk 3 above)
Comment (djkwagala) [inline src/StarRating.xml]: removed

File: src/components/StarRatingContainer.ts (hunk 5 above)
Comment (djkwagala) [inline src/components/StarRatingContainer.ts]: fixed

File: src/components/StarRatingContainer.ts (hunk 8 above)
Comment (djkwagala) [inline src/components/StarRatingContainer.ts]: fixed

File: src/components/__tests__/StarRating.spec.ts (hunk 6 above)
Comment (djkwagala) [inline src/components/__tests__/StarRating.spec.ts]: fixed

**Pull request merged**